velox/functions/sparksql/StringToMap.h

```diff
@@ -23,80 +23,105 @@ namespace facebook::velox::functions::sparksql {
 template <typename T>
 struct StringToMapFunction {
   VELOX_DEFINE_FUNCTION_TYPES(T);
+  static StringView entryDelimiter_;
+  static StringView keyValueDelimiter_;
 
   // Results refer to strings in the first argument.
   static constexpr int32_t reuse_strings_from_arg = 0;
-
+  // One parameter
+  void call(
+      out_type<Map<Varchar, Varchar>>& out,
+      const arg_type<Varchar>& input) {
+    callImpl(out, input, entryDelimiter_, keyValueDelimiter_);
+  }
+  // Two parameters
+  void call(
+      out_type<Map<Varchar, Varchar>>& out,
+      const arg_type<Varchar>& input,
+      const arg_type<Varchar>& entryDelimiter) {
+    VELOX_USER_CHECK(
+        entryDelimiter.size() >= 1, "entryDelimiter's size should >= 1.");
+    callImpl(out, input, entryDelimiter, keyValueDelimiter_);
+  }
+  // Three parameters
   void call(
       out_type<Map<Varchar, Varchar>>& out,
       const arg_type<Varchar>& input,
       const arg_type<Varchar>& entryDelimiter,
       const arg_type<Varchar>& keyValueDelimiter) {
-    VELOX_USER_CHECK_EQ(
-        entryDelimiter.size(), 1, "entryDelimiter's size should be 1.");
-    VELOX_USER_CHECK_EQ(
-        keyValueDelimiter.size(), 1, "keyValueDelimiter's size should be 1.");
-
-    callImpl(
-        out,
-        toStringView(input),
-        toStringView(entryDelimiter),
-        toStringView(keyValueDelimiter));
+    VELOX_USER_CHECK(
+        entryDelimiter.size() >= 1, "entryDelimiter's size should >= 1.");
+    VELOX_USER_CHECK(
+        keyValueDelimiter.size() >= 1, "keyValueDelimiter's size should >= 1.");
+    callImpl(out, input, entryDelimiter, keyValueDelimiter);
   }
 
  private:
-  static std::string_view toStringView(const arg_type<Varchar>& input) {
-    return std::string_view(input.data(), input.size());
+  std::vector<std::pair<int, int>> strStrRex(
```

Comment:
Why do we have this name? Please add some comments for this function.

Reply:
done